google/cloud/spanner_dbapi/connection.py

```diff
@@ -201,17 +201,14 @@ def transaction_checkout(self):
         Begin a new transaction, if there is no transaction in
         this connection yet. Return the begun one otherwise.
 
-        The method is non operational in autocommit mode.
-
         :rtype: :class:`google.cloud.spanner_v1.transaction.Transaction`
         :returns: A Cloud Spanner transaction object, ready to use.
         """
-        if not self.autocommit:
-            if not self.inside_transaction:
-                self._transaction = self._session_checkout().transaction()
-                self._transaction.begin()
+        if not self.inside_transaction:
+            self._transaction = self._session_checkout().transaction()
+            self._transaction.begin()
 
-            return self._transaction
+        return self._transaction
```

Comment:
As we're going to use transactions for batches in `autocommit` mode, this method now should also work in `autocommit` mode.